shares/parse.go

```diff
@@ -36,13 +36,7 @@ func ParseShares(shares []Share, ignorePadding bool) ([]ShareSequence, error) {
 	currentSequence := ShareSequence{}
 
 	for _, share := range shares {
-		if err := share.Validate(); err != nil {
```

Comment:
Why validation is dropped?

Reply:
Shares are made immutable (we encapsulate the []byte in a private field) and validation is done during construction so validating shares every time we call a function on it is not necessary